Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,568 @@
+{
+    IQueryable results = options.Filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+If we don't want to pass the filter query option in the request, but we want a filter to be applied, we can initialize the `FilterQueryOption` and call `ApplyTo` to apply the filter query to the data.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    var filter = new FilterQueryOption("Id eq 1", options.Context,
+        new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$filter", "Id eq 1" } },
+            options.Context.RequestContainer));
+
+    IQueryable results = filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+When making the request, we don't need to use the `$expand` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+### Apply query
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+In the example above, `options.Apply` property will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$apply=aggregate(Age with max as MaxAge)
+```
+The `options.Apply` property is an instance of `ApplyQueryOption`.
+
+Alternatively we can call `ApplyTo` on the `Apply` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Apply.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+If we don't want to pass the query option in the request, but we want to aggregate the results, we can initialize the `ApplyQueryOption` and set it in the request.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    var queryOptionParser = new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$apply", "aggregate(Age with max as MaxAge)" } },
+            options.Context.RequestContainer);
+
+    options.Request.ODataFeature().ApplyClause = new ApplyQueryOption("aggregate(Age with max as MaxAge)", options.Context, queryOptionParser).ApplyClause;
+
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+## Extending the EnableQueryAttribute class
+The `EnableQueryAttribute` class is NOT sealed. This means it can be extended.
+
+`EnableQueryAttribute` class has several virtual methods that can be overriden.
```

Comment:
You could alternatively use the doc comments on the respective methods in `EnableQueryAttribute` class to describe what it does...